lib/y2logsstep.pm

```diff
@@ -236,13 +238,24 @@ sub save_upload_y2logs {
     $suffix //= '';
 
     return if (get_var('NOLOGS'));
-
-    assert_script_run 'sed -i \'s/^tar \(.*$\)/tar --warning=no-file-changed -\1 || true/\' /usr/sbin/save_y2logs';
-    my $filename = "/tmp/y2logs$suffix.tar" . get_available_compression();
-    assert_script_run "save_y2logs $filename", 180;
-    upload_logs $filename;
-    save_screenshot();
-    $self->investigate_yast2_failure();
+    # Try to recover network if cannot reach gw
+    my $can_upload_logs = can_upload_logs;
+    unless ($can_upload_logs) {
+        $can_upload_logs = recover_network;
+    }
+    # Simply upload logs if everything works
+    if ($can_upload_logs) {
```

Comment:
if perl handle logical expressions as any other language you can delete evething above this line and replace current if with following 
`if(can_upload_logs() || recover_network() )` 
I think it will give you same result ( 1. calling can_upload_logs  at first 2. in case of failure calling recover_network 3. proceed with code inside if condition only when at least one function pass )